xdsl/printer.py

```diff
@@ -418,19 +418,19 @@ def print_attribute(self, attribute: Attribute) -> None:
         if (isinstance(attribute, DenseIntOrFPElementsAttr)
                 and self.target == self.Target.MLIR):
 
+            def print_one_elem(val: Attribute):
+                if isinstance(val, IntegerAttr):
+                    self.print(val.value.data)
+                elif isinstance(val, FloatAttr):
```

Comment:
`if isinstance(val, IntegerAttr | FloatAttr):` ? I find it more readable, maybe there are good points for your style though

Reply:
I think that won't work, you'd have to `isinstance(val, (IntegerAttr, FloatAttr))`. `isinstance` does not work with `typing` stuff